### PR TITLE
feat(service): add remote files to dataset

### PR DESCRIPTION
**Pull request by jsam**
closes: #1063

### CODE REVIEW

File: tests/service/jobs/test_datasets.py [hunk 1]
```diff
@@ -271,3 +272,55 @@ def test_dataset_import_twice_job(doi, svc_client_with_repo):
...
     assert 'Dataset exists' in extras['error']
+
+
+@pytest.mark.parametrize(
+    'url', [
+        'https://gist.github.com/jsam/d957f306ed0fe4ff018e902df6a1c8e3',
+    ]
+)
+@pytest.mark.integration
+@pytest.mark.service
```
Comment (mohammad-sdsc) [inline tests/service/jobs/test_datasets.py]: Does this need flaky as well?

File: tests/service/views/test_dataset_views.py [hunk 2]
```diff
@@ -807,3 +807,195 @@ def test_import_dataset_job_enqueue(
...
+            'dataset_name': uuid.uuid4().hex,
+            'create_dataset': True,
+            'files': [{
+                'file_url': url_gist,
+            }, {
+                'file_url': url_dbox,
+            }]
+        }),
+        headers=headers,
+    )
```
Comment (mohammad-sdsc) [inline tests/service/views/test_dataset_views.py]: This code has a lot of redundancy with the above function. Please consider to factor it to a function.

File: tests/service/jobs/test_datasets.py (hunk 1 above)
Comment (jsam) [reply tests/service/jobs/test_datasets.py:285]: done

File: tests/service/views/test_dataset_views.py (hunk 2 above)
Comment (jsam) [reply tests/service/views/test_dataset_views.py:879]: done.

**Pull request merged**